### ISSUE TITLE
triple backticks preceded by other characters should not mark the start/end of a code block

### ISSUE THREAD

**Issue by weakish** [thepracticaldev/dev.to#8044]
**Describe the bug**

dev.to mistakenly treat triple backticks preceded by other characters as indicating the start/end of a code block.

**To Reproduce**

In a dev.to post, write the following markdown:

    ```rust
    /// # Example
    ///
    /// ```
    /// let foo = "foo";
    /// assert_eq!(foo, "foo");
    /// ```
    ```

    ## unsafe

**Expected behavior**

The above markdown should rendered like below (as rendered by GitHub Pages):

![image](https://user-images.githubusercontent.com/114114/82759618-2334e580-9e21-11ea-91fc-f70e05650cc4.png)

However, dev.to rendered it as:

![image](https://user-images.githubusercontent.com/114114/82759644-49f31c00-9e21-11ea-9c01-db7bcd6e35a9.png)

**Comment by citizen428** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by citizen428** [comment]
Thanks @weakish! 

To anyone attempting to fix this: a [previous PR](https://github.com/thepracticaldev/dev.to/pull/5785) introduced a `MarkdownTraverser` class (at `app/labor/markdown_traverser.rb`), it's probably worth exploring to see how/if we can leverage that for this problem.

**Comment by citizen428** [comment]
I did some more digging and found something interesting: [this PR](https://github.com/thepracticaldev/dev.to/pull/4438/files) added the `disable_indented_code_blocks` option to our Redcarpet initializer. This was later disabled again by [this follow-up PR](https://github.com/thepracticaldev/dev.to/pull/4988), so it seems like this is intended behavior.

I changed the label on this issue from `type: bug` to `type: discussion` and would like to get some input from @maestromac, @benhalpern and others here.

**Comment by weakish** [comment]
I think this has nothing to do with `disable_indented_code_blocks`. `disable_indented_code_blocks` controls whether you can use original markdown code block (indent by four spaces) to escape triple backticks code block.

But here triple backticks is not indented by spaces, but preceded by other non space characters (as `///` in the example).

**Comment by citizen428** [comment]
Ok, thanks for the clarification, I'm not overly familiar with Redcarpet's options.
